static/js/containers/CourseIndexPage.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+import React from "react"
+import { connectRequest, querySelectors } from "redux-query"
+import { connect } from "react-redux"
+import { compose } from "redux"
+
+import CourseDrawer from "./CourseDrawer"
+
+import CourseCarousel from "../components/CourseCarousel"
+
+import { setShowCourseDrawer } from "../actions/ui"
+import {
+  featuredCoursesRequest,
+  featuredCoursesSelector,
+  upcomingCoursesRequest,
+  upcomingCoursesSelector,
+  newCoursesRequest,
+  newCoursesSelector
+} from "../lib/api/courses"
+
+import type { Course } from "../flow/discussionTypes"
+
+type OwnProps = {||}
+
+type StateProps = {|
+  featuredCourses: Array<Course>,
+  upcomingCourses: Array<Course>,
+  newCourses: Array<Course>,
+  loaded: boolean
+|}
+
+type DispatchProps = {|
+  setShowCourseDrawer: Function
+|}
+
+type Props = {|
+  ...StateProps,
+  ...OwnProps,
+  ...DispatchProps
+|}
+
+export const CourseIndexPage = ({
+  upcomingCourses,
+  featuredCourses,
+  newCourses,
+  loaded,
+  setShowCourseDrawer
+}: Props) => (
+  <React.Fragment>
+    <div className="main-content one-column">
+      {loaded ? (
+        <React.Fragment>
+          <CourseCarousel
```

Comment:
Maybe there should be a conditional to check if there are any featured courses before displaying this carousel?  Unless it's a sure thing that there will be featured courses.

<img width="979" alt="Screen Shot 2019-03-13 at 1 25 58 PM" src="https://user-images.githubusercontent.com/187676/54300713-ce96a900-4593-11e9-9d95-b6d6e3c9d112.png">

Reply:
yeah that's a good idea